### PR TITLE
tsweb: add more test cases for TestCleanRedirectURL

### PR DESCRIPTION
**Pull request by noncombatant**
Updates #cleanup

### CODE REVIEW

**Comment by noncombatant** [comment]
Regarding

```
--- FAIL: TestCleanRedirectURL (0.00s)
    tsweb_test.go:678: CleanRedirectURL("", [tailscale.com]) got "", want an error
FAIL
FAIL    tailscale.com/tsweb     1.039s
FAIL
```

, is the test wrong or the code, do you think? I think this is this test case:

```
		{"", tailscaleHost, "", true},
```

)

**Review by awly — APPROVED** [review]

File: tsweb/tsweb_test.go [hunk 1]
```diff
@@ -650,6 +649,15 @@ func TestCleanRedirectURL(t *testing.T) {
 		{"http://myserver:31544/a/foobar", myServer, "http://myserver:31544/a/foobar", false},
 		{"http://evil.com/foo", tailscaleHost, "", true},
 		{"//evil.com", tailscaleHost, "", true},
+		{"\\\\evil.com", tailscaleHost, "", true},
+		{"javascript:alert(123)", tailscaleHost, "", true},
+		{"file:///", tailscaleHost, "", true},
+		{"file:////SERVER/directory/goats.txt", tailscaleHost, "", true},
+		{"https://google.com", tailscaleHost, "", true},
+		{"", tailscaleHost, "", true},
```
Comment (awly) [inline tsweb/tsweb_test.go:657]: The test case is wrong IMO. The `allowedHosts` parameter (`tailscaleHost` in this case) is a list of all hostnames you can put in the redirect URL. But a redirect URL can be a relative path, like `/admin/machines` and without a hostname.

This specific case (empty string) means redirect to `/`, which is valid. We want to support it, in particular here: https://github.com/tailscale/corp/blob/5af8524435526561c51f15faed439c9902c9bede/control/auth/signin.go#L36 (if you go through the login flow without setting `next_url` at all).

At the very bottom, there's already a test case that handles empty URL.
```suggestion
```

Comment (noncombatant) [reply tsweb/tsweb_test.go:657]: How about, to document and enforce the behavior you describe,

```
{"", tailscaleHost, "", false},
```

? (It passes locally.)

Comment (awly) [reply tsweb/tsweb_test.go:657]: Sure!

**Pull request merged**